File: src/EyeTimeTracker.Core/Models/SyncSettings.cs
using EyeTimeTracker.Core.Sync;

namespace EyeTimeTracker.Core.Models;

public sealed class SyncSettings
{
    public static SyncSettings Unpaired => new();

    public bool IsPaired { get; set; }
    public string PeerDeviceId { get; set; } = string.Empty;
    public string PeerPlatform { get; set; } = string.Empty;
    public string SharedSecret { get; set; } = string.Empty;
    public string LastKnownHost { get; set; } = string.Empty;
    public int LastKnownPort { get; set; }
    public long LastSyncUnixSeconds { get; set; }
    public string LastError { get; set; } = string.Empty;
    public ReminderRuntimeState LocalReminderState { get; set; } = new();
    public ReminderRuntimeState PeerReminderState { get; set; } = new();
}

File: android/EyeTimeTrackerAndroid/src/com/eyetimetracker/android/SyncSettings.java
package com.eyetimetracker.android;

public final class SyncSettings {
    public boolean isPaired;
    public String peerDeviceId = "";
    public String peerPlatform = "";
    public String peerHost = "";
    public int peerPort;
    public String sharedSecret = "";
    public long lastSyncUnixSeconds;
    public String lastError = "";
    public ReminderRuntimeState peerReminderState = new ReminderRuntimeState();

    public static SyncSettings unpaired() {
        return new SyncSettings();
    }
}
